python/iceberg/api/partition_spec.py

```diff
@@ -183,7 +183,7 @@ def __repr__(self):
         return "".join(sb)
 
     @staticmethod
-    def builder_for(schema):
+    def builder_for(schema: Schema) -> "PartitionSpecBuilder":
```

Comment:
Why is `PartitionSpecBuilder` in quotes?

Not a problem, I've just not seen that before. `Schema` isn't in quotes.

Reply:
Its because this is above the `PartitionSpecBuilder` class in the file. So the `PartitionSpecBuilder` class isn't yet reference-able. One of the small annoyances of mypy. Moving the class to the top of the file or to the a new file would fix it but create a needlessly larger diff

Note: the same double-quote notation is used for a method in `PartitionSpecBuilder` which returns a `PartitionSpecBuilder` for the same reason, this is unfortunately unavoidable (to my knowledge) .